modules/src/ics02_client/query.rs

```diff
@@ -3,10 +3,34 @@ use std::marker::PhantomData;
 use crate::ics23_commitment::{CommitmentPath, CommitmentProof};
 
 //use crate::ics02_client::state::{ClientState, ConsensusState};
+use crate::ics03_connection::error::Kind;
 use crate::ics24_host::identifier::ClientId;
 use crate::path::{ClientStatePath, ConsensusStatePath};
+use crate::try_from_raw::TryFromRaw;
 use crate::Height;
 
+//TODO: This might need to be migrated to ibc-proto crate. But ClientConnections (as array of strings)
+// might not be part of an official proto file
```

Comment:
I forgot about this comment here. So, currently, it seems that the ClientConnections are not created as proto files because they are not in the provable store. This might change but until then we should keep this implementation here. The ibc-proto crate is auto-generated while this needs manual attention if something changes in the spec.

Homework: I believe that Ethan opened an issue about proto-ing this in the Cosmos SDK repo. Find the issue and replace this comment with a link to the issue. Alternatively, if you can't find the issue, open a new one. :)

Reply:
Are you talking about this [issue](https://github.com/cosmos/cosmos-sdk/issues/6736) ? I think this is for ClientState. Couldn't find anything for client connections.

Reply:
Yes, that's the one I was thinking about! But you're right, that's slightly similar but not exactly the same.

All in all we're not going to move this into ibc-proto but instead we'll wait for the SDK team to create proto files for it. (Based on the issue, currently, it's pending on Tendermint Go.) If you could update the comment to reflect that, then we can leave it here until that time comes.

Reply:
Ok, agree once Tendermint has proto types for this we can refactor and move the code to ibc-proto. So this is good for now.